LocationProvider/build.gradle

```diff
@@ -0,0 +1,39 @@
+plugins {
+    id 'com.android.library'
+}
+
+android {
+    compileSdkVersion 31
+
+    defaultConfig {
+        minSdkVersion 23
+        targetSdkVersion 31
```

Comment:
Yes you are right and I have enabled the following in `gradle.properties`:
```
android.useAndroidX=true
android.enableJetifier=true
```
As for the `targetSdkVersion` and `compileSdkVersion`, the former cannot be less than 30. I tried it out in my latest version of Android Studio (**Android Studio Bumblebee | 2021.1.1 Patch 1**) and I got a red squiggly line beneath the `targetSdkVersion 28` line telling me the following:
`Google Play requires that apps target API level 30 or higher.`
Then, I updated it to `targetSdkVersion 30` and then it got highlighted yellow with the following warning:
`Not targeting the latest versions of Android; compatibility modes apply. Consider testing and updating this version. Consult the android.os.Build.VERSION_CODES javadoc for details`
I personally always take warnings (that are highlighted yellow) in my code very seriously but if you think we should set the targetSdkVersion to 30, then I am fine with it. Of course, since the compileSdkVersion cannot be less than the targetSdkVersion, it must also be set to 30. Currently, below are the changes I made in build.gradle (app):
```
compileSdkVersion 30
targetSdkVersion 30
```

Reply:
Okay, so there seems to be a problem when I try to compile with those sdk versions:

```
> Task :app:processFdroidDebugResources FAILED
AGPBI: {"kind":"error","text":"Android resource linking failed","sources":[{"file":"/home/muhammadmehdi/.gradle/caches/transforms-2/files-2.1/454708d6096884ae40be6f0df38c8d79/material-1.5.0/res/values-v31/values-v31.xml","position":{"startLine":2,"startColumn":4,"startOffset":55,"endColumn":93,"endOffset":144}}],"original":"/home/muhammadmehdi/.gradle/caches/transforms-2/files-2.1/454708d6096884ae40be6f0df38c8d79/material-1.5.0/res/values-v31/values-v31.xml:3:5-94: AAPT: error: resource android:color/system_neutral1_1000 not found.\n    ","tool":"AAPT"}
AGPBI: {"kind":"error","text":"Android resource linking failed","sources":[{"file":"/home/muhammadmehdi/.gradle/caches/transforms-2/files-2.1/454708d6096884ae40be6f0df38c8d79/material-1.5.0/res/values-v31/values-v31.xml","position":{"startLine":3,"startColumn":4,"startOffset":149,"endColumn":93,"endOffset":238}}],"original":"/home/muhammadmehdi/.gradle/caches/transforms-2/files-2.1/454708d6096884ae40be6f0df38c8d79/material-1.5.0/res/values-v31/values-v31.xml:4:5-94: AAPT: error: resource android:color/system_neutral1_900 not found.\n    ","tool":"AAPT"}
AGPBI: {"kind":"error","text":"Android resource linking failed","sources":[{"file":"/home/muhammadmehdi/.gradle/caches/transforms-2/files-2.1/454708d6096884ae40be6f0df38c8d79/material-1.5.0/res/values-v31/values-v31.xml","position":{"startLine":4,"startColumn":4,"startOffset":243,"endColumn":92,"endOffset":331}}],"original":"/home/muhammadmehdi/.gradle/caches/transforms-2/files-2.1/454708d6096884ae40be6f0df38c8d79/material-1.5.0/res/values-v31/values-v31.xml:5:5-93: AAPT: error: resource android:color/system_neutral1_0 not found.\n    ","tool":"AAPT"}
AGPBI: {"kind":"error","text":"Android resource linking failed","sources":[{"file":"/home/muhammadmehdi/.gradle/caches/transforms-2/files-2.1/454708d6096884ae40be6f0df38c8d79/material-1.5.0/res/values-v31/values-v31.xml","position":{"startLine":5,"startColumn":4,"startOffset":336,"endColumn":93,"endOffset":425}}],"original":"/home/muhammadmehdi/.gradle/caches/transforms-2/files-2.1/454708d6096884ae40be6f0df38c8d79/material-1.5.0/res/values-v31/values-v31.xml:6:5-94: AAPT: error: resource android:color/system_neutral1_800 not found.\n    ","tool":"AAPT"}
AGPBI: {"kind":"error","text":"Android resource linking failed","sources":[{"file":"/home/muhammadmehdi/.gradle/caches/transforms-2/files-2.1/454708d6096884ae40be6f0df38c8d79/material-1.5.0/res/values-v31/values-v31.xml","position":{"startLine":6,"startColumn":4,"startOffset":430,"endColumn":93,"endOffset":519}}],"original":"/home/muhammadmehdi/.gradle/caches/transforms-2/files-2.1/454708d6096884ae40be6f0df38c8d79/material-1.5.0/res/values-v31/values-v31.xml:7:5-94: AAPT: error: resource android:color/system_neutral1_700 not found.\n    ","tool":"AAPT"}
AGPBI: {"kind":"error","text":"Android resource linking failed","sources":[{"file":"/home/muhammadmehdi/.gradle/caches/transforms-2/files-2.1/454708d6096884ae40be6f0df38c8d79/material-1.5.0/res/values-v31/values-v31.xml","position":{"startLine":7,"startColumn":4,"startOffset":524,"endColumn":93,"endOffset":613}}],"original":"/home/muhammadmehdi/.gradle/caches/transforms-2/files-2.1/454708d6096884ae40be6f0df38c8d79/material-1.5.0/res/values-v31/values-v31.xml:8:5-94: AAPT: error: resource android:color/system_neutral1_600 not found.\n    ","tool":"AAPT"}
AGPBI: {"kind":"error","text":"Android resource linking failed","sources":[{"file":"/home/muhammadmehdi/.gradle/caches/transforms-2/files-2.1/454708d6096884ae40be6f0df38c8d79/material-1.5.0/res/values-v31/values-v31.xml","position":{"startLine":8,"startColumn":4,"startOffset":618,"endColumn":93,"endOffset":707}}],"original":"/home/muhammadmehdi/.gradle/caches/transforms-2/files-2.1/454708d6096884ae40be6f0df38c8d79/material-1.5.0/res/values-v31/values-v31.xml:9:5-94: AAPT: error: resource android:color/system_neutral1_500 not found.\n    ","tool":"AAPT"}
AGPBI: {"kind":"error","text":"Android resource linking failed","sources":[{"file":"/home/muhammadmehdi/.gradle/caches/transforms-2/files-2.1/454708d6096884ae40be6f0df38c8d79/material-1.5.0/res/values-v31/values-v31.xml","position":{"startLine":9,"startColumn":4,"startOffset":712,"endColumn":93,"endOffset":801}}],"original":"/home/muhammadmehdi/.gradle/caches/transforms-2/files-2.1/454708d6096884ae40be6f0df38c8d79/material-1.5.0/res/values-v31/values-v31.xml:10:5-94: AAPT: error: resource android:color/system_neutral1_400 not found.\n    ","tool":"AAPT"}
AGPBI: {"kind":"error","text":"Android resource linking failed","sources":[{"file":"/home/muhammadmehdi/.gradle/caches/transforms-2/files-2.1/454708d6096884ae40be6f0df38c8d79/material-1.5.0/res/values-v31/values-v31.xml","position":{"startLine":10,"startColumn":4,"startOffset":806,"endColumn":93,"endOffset":895}}],"original":"/home/muhammadmehdi/.gradle/caches/transforms-2/files-2.1/454708d6096884ae40be6f0df38c8d79/material-1.5.0/res/values-v31/values-v31.xml:11:5-94: AAPT: error: resource android:color/system_neutral1_300 not found.\n    ","tool":"AAPT"}
AGPBI: {"kind":"error","text":"Android resource linking failed","sources":[{"file":"/home/muhammadmehdi/.gradle/caches/transforms-2/files-2.1/454708d6096884ae40be6f0df38c8d79/material-1.5.0/res/values-v31/values-v31.xml","position":{"startLine":11,"startColumn":4,"startOffset":900,"endColumn":93,"endOffset":989}}],"original":"/home/muhammadmehdi/.gradle/caches/transforms-2/files-2.1/454708d6096884ae40be6f0df38c8d79/material-1.5.0/res/values-v31/values-v31.xml:12:5-94: AAPT: error: resource android:color/system_neutral1_200 not found.\n    ","tool":"AAPT"}
AGPBI: {"kind":"error","text":"Android resource linking failed","sources":[{"file":"/home/muhammadmehdi/.gradle/caches/transforms-2/files-2.1/454708d6096884ae40be6f0df38c8d79/material-1.5.0/res/values-v31/values-v31.xml","position":{"startLine":12,"startColumn":4,"startOffset":994,"endColumn":93,"endOffset":1083}}],"original":"/home/muhammadmehdi/.gradle/caches/transforms-2/files-2.1/454708d6096884ae40be6f0df38c8d79/material-1.5.0/res/values-v31/values-v31.xml:13:5-94: AAPT: error: resource android:color/system_neutral1_100 not found.\n    ","tool":"AAPT"}
AGPBI: {"kind":"error","text":"Android resource linking failed","sources":[{"file":"/home/muhammadmehdi/.gradle/caches/transforms-2/files-2.1/454708d6096884ae40be6f0df38c8d79/material-1.5.0/res/values-v31/values-v31.xml","position":{"startLine":13,"startColumn":4,"startOffset":1088,"endColumn":92,"endOffset":1176}}],"original":"/home/muhammadmehdi/.gradle/caches/transforms-2/files-2.1/454708d6096884ae40be6f0df38c8d79/material-1.5.0/res/values-v31/values-v31.xml:14:5-93: AAPT: error: resource android:color/system_neutral1_50 not found.\n    ","tool":"AAPT"}
AGPBI: {"kind":"error","text":"Android resource linking failed","sources":[{"file":"/home/muhammadmehdi/.gradle/caches/transforms-2/files-2.1/454708d6096884ae40be6f0df38c8d79/material-1.5.0/res/values-v31/values-v31.xml","position":{"startLine":14,"startColumn":4,"startOffset":1181,"endColumn":92,"endOffset":1269}}],"original":"/home/muhammadmehdi/.gradle/caches/transforms-2/files-2.1/454708d6096884ae40be6f0df38c8d79/material-1.5.0/res/values-v31/values-v31.xml:15:5-93: AAPT: error: resource android:color/system_neutral1_10 not found.\n    ","tool":"AAPT"}
AGPBI: {"kind":"error","text":"Android resource linking failed","sources":[{"file":"/home/muhammadmehdi/.gradle/caches/transforms-2/files-2.1/454708d6096884ae40be6f0df38c8d79/material-1.5.0/res/values-v31/values-v31.xml","position":{"startLine":15,"startColumn":4,"startOffset":1274,"endColumn":101,"endOffset":1371}}],"original":"/home/muhammadmehdi/.gradle/caches/transforms-2/files-2.1/454708d6096884ae40be6f0df38c8d79/material-1.5.0/res/values-v31/values-v31.xml:16:5-102: AAPT: error: resource android:color/system_neutral2_1000 not found.\n    ","tool":"AAPT"}
AGPBI: {"kind":"error","text":"Android resource linking failed","sources":[{"file":"/home/muhammadmehdi/.gradle/caches/transforms-2/files-2.1/454708d6096884ae40be6f0df38c8d79/material-1.5.0/res/values-v31/values-v31.xml","position":{"startLine":16,"startColumn":4,"startOffset":1376,"endColumn":101,"endOffset":1473}}],"original":"/home/muhammadmehdi/.gradle/caches/transforms-2/files-2.1/454708d6096884ae40be6f0df38c8d79/material-1.5.0/res/values-v31/values-v31.xml:17:5-102: AAPT: error: resource android:color/system_neutral2_900 not found.\n    ","tool":"AAPT"}
AGPBI: {"kind":"error","text":"Android resource linking failed","sources":[{"file":"/home/muhammadmehdi/.gradle/caches/transforms-2/files-2.1/454708d6096884ae40be6f0df38c8d79/material-1.5.0/res/values-v31/values-v31.xml","position":{"startLine":17,"startColumn":4,"startOffset":1478,"endColumn":100,"endOffset":1574}}],"original":"/home/muhammadmehdi/.gradle/caches/transforms-2/files-2.1/454708d6096884ae40be6f0df38c8d79/material-1.5.0/res/values-v31/values-v31.xml:18:5-101: AAPT: error: resource android:color/system_neutral2_0 not found.\n    ","tool":"AAPT"}
AGPBI: {"kind":"error","text":"Android resource linking failed","sources":[{"file":"/home/muhammadmehdi/.gradle/caches/transforms-2/files-2.1/454708d6096884ae40be6f0df38c8d79/material-1.5.0/res/values-v31/values-v31.xml","position":{"startLine":18,"startColumn":4,"startOffset":1579,"endColumn":101,"endOffset":1676}}],"original":"/home/muhammadmehdi/.gradle/caches/transforms-2/files-2.1/454708d6096884ae40be6f0df38c8d79/material-1.5.0/res/values-v31/values-v31.xml:19:5-102: AAPT: error: resource android:color/system_neutral2_800 not found.\n    ","tool":"AAPT"}
AGPBI: {"kind":"error","text":"Android resource linking failed","sources":[{"file":"/home/muhammadmehdi/.gradle/caches/transforms-2/files-2.1/454708d6096884ae40be6f0df38c8d79/material-1.5.0/res/values-v31/values-v31.xml","position":{"startLine":19,"startColumn":4,"startOffset":1681,"endColumn":101,"endOffset":1778}}],"original":"/home/muhammadmehdi/.gradle/caches/transforms-2/files-2.1/454708d6096884ae40be6f0df38c8d79/material-1.5.0/res/values-v31/values-v31.xml:20:5-102: AAPT: error: resource android:color/system_neutral2_700 not found.\n    ","tool":"AAPT"}
AGPBI: {"kind":"error","text":"Android resource linking failed","sources":[{"file":"/home/muhammadmehdi/.gradle/caches/transforms-2/files-2.1/454708d6096884ae40be6f0df38c8d79/material-1.5.0/res/values-v31/values-v31.xml","position":{"startLine":20,"startColumn":4,"startOffset":1783,"endColumn":101,"endOffset":1880}}],"original":"/home/muhammadmehdi/.gradle/caches/transforms-2/files-2.1/454708d6096884ae40be6f0df38c8d79/material-1.5.0/res/values-v31/values-v31.xml:21:5-102: AAPT: error: resource android:color/system_neutral2_600 not found.\n    ","tool":"AAPT"}
AGPBI: {"kind":"error","text":"Android resource linking failed","sources":[{"file":"/home/muhammadmehdi/.gradle/caches/transforms-2/files-2.1/454708d6096884ae40be6f0df38c8d79/material-1.5.0/res/values-v31/values-v31.xml","position":{"startLine":21,"startColumn":4,"startOffset":1885,"endColumn":101,"endOffset":1982}}],"original":"/home/muhammadmehdi/.gradle/caches/transforms-2/files-2.1/454708d6096884ae40be6f0df38c8d79/material-1.5.0/res/values-v31/values-v31.xml:22:5-102: AAPT: error: resource android:color/system_neutral2_500 not found.\n    ","tool":"AAPT"}

> Task :app:checkFdroidDebugDuplicateClasses UP-TO-DATE
> Task :app:desugarFdroidDebugFileDependencies
> Task :app:mergeExtDexFdroidDebug UP-TO-DATE
> Task :app:mergeFdroidDebugShaders UP-TO-DATE
> Task :app:compileFdroidDebugShaders UP-TO-DATE
> Task :app:generateFdroidDebugAssets UP-TO-DATE
> Task :LocationProvider:mergeDebugShaders UP-TO-DATE
> Task :LocationProvider:compileDebugShaders UP-TO-DATE
> Task :LocationProvider:generateDebugAssets UP-TO-DATE
> Task :LocationProvider:packageDebugAssets UP-TO-DATE
> Task :app:mergeFdroidDebugAssets UP-TO-DATE
> Task :app:processFdroidDebugJavaRes NO-SOURCE
> Task :LocationProvider:processDebugJavaRes NO-SOURCE
> Task :LocationProvider:bundleLibResDebug UP-TO-DATE
> Task :LocationProvider:bundleLibRuntimeDebug UP-TO-DATE
> Task :app:validateSigningFdroidDebug UP-TO-DATE
> Task :app:signingConfigWriterFdroidDebug UP-TO-DATE
> Task :app:mergeLibDexFdroidDebug UP-TO-DATE
> Task :app:mergeFdroidDebugJniLibFolders UP-TO-DATE
> Task :LocationProvider:mergeDebugJniLibFolders UP-TO-DATE
> Task :LocationProvider:mergeDebugNativeLibs UP-TO-DATE
> Task :LocationProvider:stripDebugDebugSymbols UP-TO-DATE
> Task :LocationProvider:transformNativeLibsWithIntermediateJniLibsForDebug UP-TO-DATE
> Task :app:mergeFdroidDebugNativeLibs UP-TO-DATE
> Task :app:stripFdroidDebugDebugSymbols UP-TO-DATE

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:processFdroidDebugResources'.
> A failure occurred while executing com.android.build.gradle.internal.tasks.Workers$ActionFacade
   > Android resource linking failed
     /home/muhammadmehdi/.gradle/caches/transforms-2/files-2.1/454708d6096884ae40be6f0df38c8d79/material-1.5.0/res/values-v31/values-v31.xml:3:5-94: AAPT: error: resource android:color/system_neutral1_1000 not found.
         
     /home/muhammadmehdi/.gradle/caches/transforms-2/files-2.1/454708d6096884ae40be6f0df38c8d79/material-1.5.0/res/values-v31/values-v31.xml:4:5-94: AAPT: error: resource android:color/system_neutral1_900 not found.
         
     /home/muhammadmehdi/.gradle/caches/transforms-2/files-2.1/454708d6096884ae40be6f0df38c8d79/material-1.5.0/res/values-v31/values-v31.xml:5:5-93: AAPT: error: resource android:color/system_neutral1_0 not found.
         
     /home/muhammadmehdi/.gradle/caches/transforms-2/files-2.1/454708d6096884ae40be6f0df38c8d79/material-1.5.0/res/values-v31/values-v31.xml:6:5-94: AAPT: error: resource android:color/system_neutral1_800 not found.
         
     /home/muhammadmehdi/.gradle/caches/transforms-2/files-2.1/454708d6096884ae40be6f0df38c8d79/material-1.5.0/res/values-v31/values-v31.xml:7:5-94: AAPT: error: resource android:color/system_neutral1_700 not found.
         
     /home/muhammadmehdi/.gradle/caches/transforms-2/files-2.1/454708d6096884ae40be6f0df38c8d79/material-1.5.0/res/values-v31/values-v31.xml:8:5-94: AAPT: error: resource android:color/system_neutral1_600 not found.
         
     /home/muhammadmehdi/.gradle/caches/transforms-2/files-2.1/454708d6096884ae40be6f0df38c8d79/material-1.5.0/res/values-v31/values-v31.xml:9:5-94: AAPT: error: resource android:color/system_neutral1_500 not found.
         
     /home/muhammadmehdi/.gradle/caches/transforms-2/files-2.1/454708d6096884ae40be6f0df38c8d79/material-1.5.0/res/values-v31/values-v31.xml:10:5-94: AAPT: error: resource android:color/system_neutral1_400 not found.
         
     /home/muhammadmehdi/.gradle/caches/transforms-2/files-2.1/454708d6096884ae40be6f0df38c8d79/material-1.5.0/res/values-v31/values-v31.xml:11:5-94: AAPT: error: resource android:color/system_neutral1_300 not found.
         
     /home/muhammadmehdi/.gradle/caches/transforms-2/files-2.1/454708d6096884ae40be6f0df38c8d79/material-1.5.0/res/values-v31/values-v31.xml:12:5-94: AAPT: error: resource android:color/system_neutral1_200 not found.
         
     /home/muhammadmehdi/.gradle/caches/transforms-2/files-2.1/454708d6096884ae40be6f0df38c8d79/material-1.5.0/res/values-v31/values-v31.xml:13:5-94: AAPT: error: resource android:color/system_neutral1_100 not found.
         
     /home/muhammadmehdi/.gradle/caches/transforms-2/files-2.1/454708d6096884ae40be6f0df38c8d79/material-1.5.0/res/values-v31/values-v31.xml:14:5-93: AAPT: error: resource android:color/system_neutral1_50 not found.
         
     /home/muhammadmehdi/.gradle/caches/transforms-2/files-2.1/454708d6096884ae40be6f0df38c8d79/material-1.5.0/res/values-v31/values-v31.xml:15:5-93: AAPT: error: resource android:color/system_neutral1_10 not found.
         
     /home/muhammadmehdi/.gradle/caches/transforms-2/files-2.1/454708d6096884ae40be6f0df38c8d79/material-1.5.0/res/values-v31/values-v31.xml:16:5-102: AAPT: error: resource android:color/system_neutral2_1000 not found.
         
     /home/muhammadmehdi/.gradle/caches/transforms-2/files-2.1/454708d6096884ae40be6f0df38c8d79/material-1.5.0/res/values-v31/values-v31.xml:17:5-102: AAPT: error: resource android:color/system_neutral2_900 not found.
         
     /home/muhammadmehdi/.gradle/caches/transforms-2/files-2.1/454708d6096884ae40be6f0df38c8d79/material-1.5.0/res/values-v31/values-v31.xml:18:5-101: AAPT: error: resource android:color/system_neutral2_0 not found.
         
     /home/muhammadmehdi/.gradle/caches/transforms-2/files-2.1/454708d6096884ae40be6f0df38c8d79/material-1.5.0/res/values-v31/values-v31.xml:19:5-102: AAPT: error: resource android:color/system_neutral2_800 not found.
         
     /home/muhammadmehdi/.gradle/caches/transforms-2/files-2.1/454708d6096884ae40be6f0df38c8d79/material-1.5.0/res/values-v31/values-v31.xml:20:5-102: AAPT: error: resource android:color/system_neutral2_700 not found.
         
     /home/muhammadmehdi/.gradle/caches/transforms-2/files-2.1/454708d6096884ae40be6f0df38c8d79/material-1.5.0/res/values-v31/values-v31.xml:21:5-102: AAPT: error: resource android:color/system_neutral2_600 not found.
         
     /home/muhammadmehdi/.gradle/caches/transforms-2/files-2.1/454708d6096884ae40be6f0df38c8d79/material-1.5.0/res/values-v31/values-v31.xml:22:5-102: AAPT: error: resource android:color/system_neutral2_500 not found.
         

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 0s
40 actionable tasks: 3 executed, 37 up-to-date
```
**Setting both compileSdkVersion and targetSdkVersion to 31 removes this error**. I deleted .gradle and even tried invalidating cache and restarting but I am still getting the above error for compileSdkVersion and targetSdkVersion as 30

Reply:
Ok. I guess we should use targetSdkVersion 31.

Reply:
I will review your changes in the next two days. With a quick glance, I think android.enableJetifier=true is recommended to be removed by [Android doc](https://developer.android.com/jetpack/androidx/migrate).